### ISSUE TITLE
bug(@angular/cdk/drag-drop): Grabbing a cdkDrag element with underlying link element cause the link to be grabbed with css transition

### ISSUE THREAD

**Issue by Raphyyy** [angular/components#24403]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

When grabbing an element that just have been dropped, you will get the underlying link to be grabbed instead of the cdkDrag element if you drag from the link.
It only appears if used CSS `transition` on `transform` to make smoother moves, so I think the issue is happening when the CSS animation has not ended.
Attaching cdkDrag to a parent element instead of setting it on the <a> element doesn't change a thing. (see difference between the "To do" cdkDropList and the "Done" cdkDropList.

![grab_bug](https://user-images.githubusercontent.com/13983107/153732525-c3be1b6e-323e-4ea1-a4cd-24243a19c082.gif)

Maybe related issue : https://stackoverflow.com/questions/68924215/cdkdrag-sometimes-grabs-uderlying-link-element

Note : the attached stackblitz run Angular 12.0.0, but the issue also appears on my local with Angular 13.2.2.

### Reproduction

Steps to reproduce:
1. https://stackblitz.com/edit/angular-pqknrn-x92xnv?file=src/app/cdk-drag-drop-connected-sorting-group-example.html
2. Grab an element, drop it somewhere and quickly regrab-it from the <a> link

### Expected Behavior

The underlying link should not be dragged

### Actual Behavior

The underlying link is dragged instead of the cdkDrag element

### Environment

- Angular: 13.2.2
- CDK/Material: 13.2.2
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): Windows 11

**Comment by Raphyyy** [comment]
It looks like the issue can also appears sometimes without having to drop and redrag the same item immediately. 
Removing the css `transition` also seems to fix it

![grab_bug](https://user-images.githubusercontent.com/13983107/153733599-69614653-be5f-49f2-916c-a53c2e6dfd34.gif)

**Comment by crisbeto** [comment]
Yes, I can't reproduce it at all with the 250ms transition on either Chrome or Firefox. I could reproduce it if I bump up the transition to 2500ms, but the `pointer-events: none` on the placeholder fixed it for me. The only other cause I can see is that by default we have some logic which makes it so that you have to drag at least 5px before dragging starts so that dragging isn't mistaken for clicking. You can try to reduce it by adding the following to the component:

```
providers: [{
  provide: CDK_DRAG_CONFIG,
  dragStartThreshold: 0
}]
```

**Comment by crisbeto** [comment]
I'm testing on a Windows machine as well so it's weird that it doesn't happen for me. One more thing worth trying is to expand the CSS from above to:

```css
.cdk-drag-placeholder {
  pointer-events: none;
  touch-action: none;
  -webkit-user-drag: none;
  use-select: none;
}
```

This is what we set on other elements in order to prevent native dragging interactions.

**Comment by Raphyyy** [comment]
Ok so it looks like the issue happens when you mousedown the link on the .cdk-drag-preview while it is still moving, wait for the transition to end, and then start moving the mouse. 
![grab_bug](https://user-images.githubusercontent.com/13983107/153782003-54f31b7f-97dd-4382-876a-12ed84b9b64a.gif)
The yellow dot tells when I hold left click
That's why you might missed it with 2500ms

**Comment by crisbeto** [comment]
Thanks, I was able to reproduce it now, albeit not consistently. I've pushed another fix to #24404. You can try it out by adding `(dragstart)="$event.preventDefault()"` to the `cdkDrag` elements.

**Comment by Raphyyy** [comment]
Nice thanks @crisbeto, works great on my side
